specification/containerservice/resource-manager/Microsoft.ContainerService/aks/stable/2025-03-01/managedClusters.json

```
@@ -5194,6 +5194,23 @@
         "disableRunCommand": {
           "type": "boolean",
           "description": "Whether to disable run command for the cluster or not."
+        },
+        "enableVnetIntegration": {
+          "type": "boolean",
+          "description": "Whether to enable apiserver vnet integration for the cluster or not. See aka.ms/AksVnetIntegration for more details."
+        },
+        "subnetId": {
+          "type": "string",
+          "format": "arm-id",
+          "x-ms-arm-id-details": {
+            "allowedResources": [
+              {
+                "type": "Microsoft.Network/virtualNetworks/subnets"
+              }
+            ]
+          },
+          "title": "The subnet to be used when apiserver vnet integration is enabled.",
+          "description": "It is required when: 1. creating a new cluster with BYO Vnet; 2. updating an existing cluster to enable apiserver vnet integration."
```

Might be cleaner to just phrase this like:
```suggestion
          "description": "It is required when creating a new cluster with BYO Vnet, or when updating an existing cluster to enable apiserver vnet integration."
```